src/main/webapp/js/upload.js

```diff
@@ -33,3 +42,15 @@ async function fetchBlobstoreUrlAndShowForm() {
   // The form is hidden by default.
   uploadForm.classList.remove('hidden');
 }
+
+/**
+ * Displays an error message if the user did not upload a JPEG file.
```

Comment:
Could you clarify how this works? Why is the error in the query string parameters?

Reply:
Initially, if the user submitted the form with a non-JPEG file, the servlet would redirect back to the upload page with an error flag in the URL parameter indicating to show a popup. 

I have now updated the logic to fetch the servlet response and redirect to the home page from the client-side. I added restrictions on the file type accepted on the front-end. If those are bypassed and the form is submitted with a non-JPEG file, the servlet will respond with an error code and a popup message will be shown.